File: diet-tracker/java/src/main/java/com/diet/model/UserProfile.java
package com.diet.model;

import jakarta.persistence.*;

@Entity
@Table(name = "user_profiles")
public class UserProfile {
    @Id
    private String userId;

    private double weight;
    private double height;
    private int age;
    private String gender; // M / F

    public UserProfile() {}

    public UserProfile(String userId, double weight, double height, int age, String gender) {
        this.userId = userId;
        this.weight = weight;
        this.height = height;
        this.age = age;
        this.gender = gender;
    }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }
    public double getWeight() { return weight; }
    public void setWeight(double weight) { this.weight = weight; }
    public double getHeight() { return height; }
    public void setHeight(double height) { this.height = height; }
    public int getAge() { return age; }
    public void setAge(int age) { this.age = age; }
    public String getGender() { return gender; }
    public void setGender(String gender) { this.gender = gender; }
}

File: diet-tracker/nodejs/models/userProfile.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('UserProfile', {
    userId: { type: DataTypes.STRING, primaryKey: true, field: 'user_id' },
    weight: { type: DataTypes.FLOAT, allowNull: false },
    height: { type: DataTypes.FLOAT, allowNull: false },
    age: { type: DataTypes.INTEGER, allowNull: false },
    gender: { type: DataTypes.STRING, allowNull: false }, // M / F
  }, { tableName: 'user_profiles', timestamps: false });
};
